core/components/molecules/dialog/dialog.js

```diff
@@ -34,75 +35,174 @@ const createButtonForAction = (action, index) => {
   )
 }
 
-const Dialog = props => (
-  <Overlay {...props}>
-    <DialogElement width={props.width} {...Automation('dialog')}>
-      <DialogTitleBar {...Automation('dialog.title')}>
-        <span>{props.title}</span>
-        <Link onClick={props.onClose}>
-          <Icon name="close" size={16} />
-        </Link>
-      </DialogTitleBar>
-      <DialogBody {...Automation('dialog.body')}>{props.children}</DialogBody>
-      <DialogFooter {...Automation('dialog.footer')}>
-        <ButtonGroup>{props.actions.map(createButtonForAction)}</ButtonGroup>
-      </DialogFooter>
-    </DialogElement>
-  </Overlay>
-)
-
-const DialogElement = styled.div`
+const focusOnFormInput = ({ current }) => {
+  const node = ReactDOM.findDOMNode(current)
+  const form = node.querySelector('form')
+  if (!form) return
+
+  const firstInput = form.querySelector('input')
+  if (!firstInput) return
+
+  firstInput.focus()
+}
+
+const getAccessibilityRole = (props, requiredRole, propObject) =>
```

Comment:
Please expand this function to be more easier to understand and add comments for what it does

Reply:
Done

Reply:
comment should explain *why* not *what* 😢

Reply:
np, added the 'why' as well :)